odrivemotor/src/odriveCAN/odriveCAN.py

```diff
@@ -119,46 +122,42 @@ def reconfigure(self, config: ComponentConfig, dependencies: Mapping[ResourceNam
+        if abs(power) < 0.001:
+            LOGGER.error("Cannot move motor at a power percent that is nearly 0")
         torque = power*self.current_limit*self.torque_constant
         await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
         await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
         await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x01, 'Input_Mode': 0x01})
         await self.send_can_message('Set_Input_Torque', {'Input_Torque': torque})
 
     async def go_for(self, rpm: float, revolutions: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        if abs(rpm) < 0.001:
+            LOGGER.error("Cannot move motor at an RPM that is nearly 0")
         rps = rpm / MINUTE_TO_SECOND
+        await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x03, 'Input_Mode': 0x05})
+        await self.send_can_message('Set_Traj_Vel_Limit', {'Traj_Vel_Limit': abs(rps)})
+        await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
+        await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
 
-        if revolutions == 0.0:
-            await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x02, 'Input_Mode': 0x01})
-            await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
-            await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
-            await self.send_can_message('Set_Input_Vel', {'Input_Vel': rps, 'Input_Torque_FF': 0})
-
+        current_position = await self.get_position()
+        goal_position = 0
+        if rpm > 0.0:
+            goal_position = current_position+revolutions+self.offset
+            await self.send_can_message('Set_Input_Pos', {'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
         else:
-            await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x03, 'Input_Mode': 0x05})
-            await self.send_can_message('Set_Traj_Vel_Limit', {'Traj_Vel_Limit': abs(rps)})
-            await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
-            await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
-
-            current_position = await self.get_position()
-            goal_position = 0
-            if rpm > 0.0:
-                goal_position = current_position+revolutions+self.offset
-                await self.send_can_message('Set_Input_Pos', {'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
-            else:
-                goal_position = current_position-revolutions+self.offset
-                await self.send_can_message('Set_Input_Pos', {'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
-
-            self.goal["position"] = goal_position
-            self.goal["active"] = True
+            goal_position = current_position-revolutions+self.offset
```

Comment:
We can use python's `math.copysign(a,b)` to simplify this logic a little. Can do in a further pr or now, after testing, if you want.

Reply:
I'm not sure I understand how we can use it, `math.copysign` doesn't do any addition or subtraction

Reply:
```python
            goal_position = current_position+math.copysign(revolutions+self.offset, rpm)
```

Reply:
I may have switched rpm and the added revolutions order as inputs to copysign, please check.

Reply:
And that line becomes your only calculation with a check for the edge case of rpm not equalling zero for the GoFor blocking call.

Reply:
Would also need to be updated in the serial model.

Reply:
ah okay I see. sure I can add this

Reply:
added, tested, and merging!